pkg/updatestrategy/aws_asg.go

```diff
@@ -193,14 +193,14 @@ func (n *ASGNodePoolsBackend) Scale(_ context.Context, nodePool *api.NodePool, r
 	}
 
 	for _, asg := range asgs {
-		min := int64(math.Min(float64(aws.Int64Value(asg.DesiredCapacity)), float64(aws.Int64Value(asg.MinSize))))
-		max := int64(math.Max(float64(aws.Int64Value(asg.DesiredCapacity)), float64(aws.Int64Value(asg.MaxSize))))
+		minSize := int64(math.Min(float64(aws.Int64Value(asg.DesiredCapacity)), float64(aws.Int64Value(asg.MinSize))))
+		maxSize := int64(math.Max(float64(aws.Int64Value(asg.DesiredCapacity)), float64(aws.Int64Value(asg.MaxSize))))
```

Comment:
There are min/max builtins in go since 1.21 https://pkg.go.dev/builtin#min
so something like this
```go
minSize := min(aws.Int64Value(asg.DesiredCapacity), aws.Int64Value(asg.MinSize))
```
should work